### PR TITLE
[MWPW 128657] Template X: Holiday Block 

### PR DESCRIPTION
**Pull request by toka1300**
Please always provide the [JIRA issue(s)](https://jira.corp.adobe.com/secure/RapidBoard.jspa?rapidView=34618) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix https://jira.corp.adobe.com/browse/MWPW-128657

Updated the holiday block, with enhancements and to be compatible with Template-X:

Test URLs:

- Before: 
  - [Template X full-width horizontal](https://main--express--adobecom.hlx.page/drafts/casey/template-x/holiday-collab?lighthouse=on)
- After: 
  - [Template X with Holiday block](https://mwpw-128657-2--express--adobecom.hlx.page/drafts/casey/template-x/holiday-collab?lighthouse=on)
  - [Template X with Holiday block (animated)](https://mwpw-128657-2--express--adobecom.hlx.page/drafts/casey/template-x/holiday-animated?lighthouse=on)
  - [Template X with Animated Holiday block (custom background)](https://mwpw-128657-2--express--adobecom.hlx.page/drafts/casey/template-x/holiday-custom-color?lighthouse=on)

### CODE REVIEW

**Review by apganapa — APPROVED** [review]

**Review by vhargrave — CHANGES_REQUESTED** [review]
Found a few things. Additionally, there's no hover effect when hovering over the picture part of a template. That should probably be fixed

I've also seen that a request for the image is being sent every time i hover over an image. That seems quite wrong to me.

**Comment by toka1300** [comment]
> Found a few things. Additionally, there's no hover effect when hovering over the picture part of a template. That should probably be fixed

File: express/blocks/template-x/template-x.js [hunk 1]
```diff
@@ -1089,6 +1105,146 @@ async function decorateToolbar(block, props) {
   }
 }
 
+function setAttributes(element, attributes) {
+  Object.keys(attributes).forEach((key) => element.setAttribute(key, attributes[key]));
+}
+
+function addBackgroundAnimation(block, animationUrl) {
+  const parent = block.closest('.template-x-wrapper.fullwidth.holiday');
```
[the remark this answers is not in the document]
Comment (toka1300) [reply express/blocks/template-x/template-x.js:1113]: Updated

File: express/blocks/template-x/template-x.js [hunk 2]
```diff
@@ -194,18 +202,26 @@ function constructProps(block) {
...
+        const holidayIcon = cols[2].querySelector('picture');
+        const backgroundAnimation = cols[4].textContent.trim().toLowerCase();
+
+        props.holidayBlock = true;
+        props.holidayIcon = holidayIcon || null;
+        if (backgroundColor) {
+          props.backgroundColor = backgroundColor;
+        }
+        props.backgroundAnimation = backgroundAnimation || null;
+        const contrastingTextColor = isDarkOverlayReadable(backgroundColor) ? '#242424' : '#fff';
```
[the remark this answers is not in the document]
Comment (toka1300) [reply express/blocks/template-x/template-x.js:222]: This code is only called when it is authored to be a holiday block, so it is only affecting the text within the holiday variant, which should all be the contrasting colour.

File: express/blocks/template-x/template-x.js [hunk 3]
```diff
@@ -1089,6 +1105,146 @@ async function decorateToolbar(block, props) {
...
+  });
+}
+
+function initToggleHoliday(block) {
+  const toggleBar = block.querySelector('.toggle-bar');
+  const aTag = toggleBar.querySelector('a');
+  const chev = toggleBar.querySelector('.toggle-button-chev');
+  const mobileChev = block.querySelector('.toggle-button.mobile');
+
+  aTag.addEventListener('click', (e) => {
```
[the remark this answers is not in the document]
Comment (toka1300) [reply express/blocks/template-x/template-x.js:1146]: This is to stop the propagation of the collapsing/expanding of the block and allow for the redirect

File: express/blocks/template-x/template-x.css [hunk 4]
```diff
@@ -2257,59 +2371,112 @@ main .template-x.collaboration .carousel-container {
...
+        font-size: 14px;
+    }
+
+    main .template-x.horizontal.holiday .toggle-bar h4 {
+        font-size: 15px;
+    }
+}
+
+@media (max-width: 900px) {
+    main .template-x.horizontal.holiday .toggle-bar p {
```
[the remark this answers is not in the document]
Comment (vhargrave) [reply express/blocks/template-x/template-x.css:2425]: yes please!

File: express/blocks/template-x/template-x.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vhargrave) [reply express/blocks/template-x/template-x.js:222]: ok, you should probably steer all that by setting an appropriate class though.

File: express/blocks/template-x/template-x.js [hunk 5]
```diff
@@ -1089,6 +1105,146 @@ async function decorateToolbar(block, props) {
...
+
+  topElements.append(heading);
+  toggle.append(link, toggleChev);
+  const mobileToggle = toggle.cloneNode(true);
+  mobileToggle.classList.add('mobile');
+  linkWrapper.remove();
+  bottomElements.append(subheading, toggle);
+  toggleBar.append(topElements, bottomElements, toggle);
+  block.append(mobileToggle);
+  heading.style.color = props.textColor;
```
[the remark this answers is not in the document]
Comment (toka1300) [reply express/blocks/template-x/template-x.js:1196]: Done!

File: express/blocks/template-x/template-x.js [hunk 6]
```diff
@@ -1089,6 +1105,146 @@ async function decorateToolbar(block, props) {
...
+      carouselFaderLeft.style.backgroundImage = 'none';
+    } else {
+      carouselFaderRight.style.backgroundImage = `linear-gradient(to right, rgba(0, 255, 255, 0), ${props.backgroundColor}`;
+      carouselFaderLeft.style.backgroundImage = `linear-gradient(to left, rgba(0, 255, 255, 0), ${props.backgroundColor}`;
+    }
+  }
+
+  topElements.append(heading);
+  toggle.append(link, toggleChev);
+  const mobileToggle = toggle.cloneNode(true);
```
[the remark this answers is not in the document]
Comment (toka1300) [reply express/blocks/template-x/template-x.js]: Refactored to have one toggle element

**Comment by qiyundai** [comment]
@toka1300 Found a small regression:
https://mwpw-128657-2--express--adobecom.hlx.page/drafts/docs/blocks-how-to/template-x 
The normal horizontal fullwidth one seems to no longer be fullwidth anymore

**Comment by toka1300** [comment]
Was not intentionally closed, I think the force push automatically closed it.

**Comment by toka1300** [comment]
Opening again

**Comment by toka1300** [comment]
Have moved over to this [PR](https://github.com/adobecom/express/pull/66)

**Pull request closed without merging**